test/end-to-end/application.js

```diff
@@ -0,0 +1,115 @@
+const companyApplicationConstants = require("../../src/models/constants/CompanyApplication");
+const accountConstants = require("../../src/models/constants/Account");
+
+describe("Company application endpoint test", () => {
+    describe("POST /application", () => {
+        describe("Input Validation (unsuccessful application)", () => {
+            const EndpointValidatorTester = ValidatorTester((params) => request().post("/apply/company").send(params));
+            const BodyValidatorTester = EndpointValidatorTester("body");
+            describe("email", () => {
+                const FieldValidatorTester = BodyValidatorTester("email");
+                FieldValidatorTester.isRequired();
+                FieldValidatorTester.mustBeEmail();
+            });
+
+            describe("password", () => {
+                const FieldValidatorTester = BodyValidatorTester("password");
+                FieldValidatorTester.isRequired();
+                FieldValidatorTester.mustBeString();
+                FieldValidatorTester.hasMinLength(accountConstants.password.min_length);
+                FieldValidatorTester.hasNumber();
+            });
+
+            describe("motivation", () => {
+                const FieldValidatorTester = BodyValidatorTester("motivation");
+                FieldValidatorTester.isRequired();
+                FieldValidatorTester.mustBeString();
+                FieldValidatorTester.hasMinLength(companyApplicationConstants.motivation.min_length);
+                FieldValidatorTester.hasMaxLength(companyApplicationConstants.motivation.max_length);
+            });
+
+            describe("companyName", () => {
+                const FieldValidatorTester = BodyValidatorTester("companyName");
+                FieldValidatorTester.isRequired();
+                FieldValidatorTester.mustBeString();
+                FieldValidatorTester.hasMinLength(companyApplicationConstants.companyName.min_length);
+                FieldValidatorTester.hasMaxLength(companyApplicationConstants.companyName.max_length);
+            });
+        });
+
+        describe("Without any existing application and accounts", () => {
+            beforeAll(async () => {
+                await CompanyApplication.deleteMany({});
+                await Account.deleteMany({});
+            });
+
+            const RealDateNow = Date.now;
+            const mockCurrentDate = new Date("2019-11-23");
+
+            beforeEach(() => {
+                Date.now = () => mockCurrentDate.getTime();
+            });
+
+            afterEach(() => {
+                Date.now = RealDateNow;
+            });
+
+            test("Valid creation", async () => {
+                const application = {
+                    email: "test@test.com",
+                    password: "password123",
+                    companyName: "Testing company",
+                    motivation: "This comapny has a very valid motivation, because otherwise the tests would not exist.",
```

Comment:
```suggestion
                    motivation: "This company has a very valid motivation because otherwise, the tests would not exist.",
```

Text fixes - since other stuff is getting fixed anyway, I get to be annoying :smile: